video/out/wayland_common.c

```diff
@@ -948,22 +948,45 @@ static const struct xdg_toplevel_listener xdg_toplevel_listener = {
 #endif
 };
 
+static const char *zxdg_decoration_mode_to_str(const uint32_t mode)
+{
+    switch (mode) {
+    case ZXDG_TOPLEVEL_DECORATION_V1_MODE_SERVER_SIDE:
+        return "server-side";
+    case ZXDG_TOPLEVEL_DECORATION_V1_MODE_CLIENT_SIDE:
+        return "client-side";
+    default:
+        return "<unknown>";
+    }
+}
+
 static void configure_decorations(void *data,
                                   struct zxdg_toplevel_decoration_v1 *xdg_toplevel_decoration,
                                   uint32_t mode)
 {
     struct vo_wayland_state *wl = data;
     struct mp_vo_opts *opts = wl->vo_opts;
 
-    if (wl->requested_decoration == mode)
+    if (wl->requested_decoration) {
+        if (mode != wl->requested_decoration) {
+            MP_MSG(wl, wl->warned_of_mismatch ? MSGL_DEBUG : MSGL_WARN,
+                   "Requested %s decorations but compositor responded with %s",
```

Comment:
decoration / decorations?

Reply:
"server-side decorations...", I think they are a multiple so that wording should be OK.